keyvault/client/src/main/java/com/azure/keyvault/SecretClient.java

```diff
@@ -0,0 +1,478 @@
+     * This operation requires the secrets/backup permission.
+     *
+     * @param name The name of the secret.
+     * @throws NullPointerException if {@code name} is {@code null}.
+     * @return A {@link Response} whose {@link Response#value()} contains the backed up secret blob.
+     * @throws com.azure.common.exception.ServiceRequestException when a secret with {@code name} doesn't exist in the key vault.
+     */
+    public Response<byte[]> backupSecret(String name) {
+        Objects.requireNonNull(name, "The Secret name cannot be null.");
+        return service.backupSecret(vaultEndpoint, name, API_VERSION, ACCEPT_LANGUAGE, CONTENT_TYPE_HEADER_VALUE)
+                .flatMap(base64URLResponse ->  Mono.just(new SimpleResponse<byte[]>(base64URLResponse.request(),
+                            base64URLResponse.statusCode(), base64URLResponse.headers(), base64URLResponse.value().value()))).block();
+    }
+
+    /**
+     * Restores a backed up secret, and all its versions, to a vault.
+     * This operation requires the secrets/restore permission.
+     *
+     * @param backup The backup blob associated with the secret.
+     * @throws NullPointerException if {@code SecretBackup} is {@code null}.
+     * @return A {@link Response} whose {@link Response#value()} contains the restored {@link Secret}.
+     * @throws com.azure.common.exception.ServiceRequestException when the {@code backup} is corrupted.
+     */
+    public Response<Secret> restoreSecret(byte[] backup) {
+        Objects.requireNonNull(backup, "The Secret backup parameter cannot be null.");
+        SecretRestoreRequestParameters parameters = new SecretRestoreRequestParameters().secretBackup(backup);
+        return service.restoreSecret(vaultEndpoint, API_VERSION, ACCEPT_LANGUAGE, parameters, CONTENT_TYPE_HEADER_VALUE).block();
+    }
+
+    /**
+     * List the secrets in the key vault. The list Secrets operation is applicable to the entire vault. The individual secret response
+     * in the list is represented by {@link SecretAttributes} as only the base secret identifier and its attributes are
+     * provided in the response. The secret values and individual secret versions are not listed in the response. This operation requires the secrets/list permission.
+     *
+     * <p>It is possible to get full Secrets with values from this information. Loop over the {@link SecretAttributes secretAttributes} and
+     * call {@link SecretClient#getSecret(String secretName)} . This will return the {@link Secret} secrets with values included of its latest version.</p>
+     * <pre>
+     *   List<SecretAttributes> secretAttributes = secretClient.listSecrets();
+     *   List<Secret> secrets = new ArrayList();
+     *   for (SecretAttributes secretAttr : secretAttributes) {
+     *       secrets.add(secretClient.getSecret(secretAttr.name()).value());
+     *   }
+     * </pre>
```

Comment:
Two things.

1. There should be a `getSecret` method that takes a `SecretAttribute`.
2. You should rewrite this code using Streams. Something like

```java
secretClient.listSecrets().stream()
    .map(secretClient::getSecret)
    .forEach(...)
```

Reply:
updated